mobile/src/people/people-company.tsx

```diff
@@ -1,23 +1,144 @@
 import React from 'react';
 import { Action } from 'redux';
-import { connect, Dispatch } from 'react-redux';
-import { View } from 'react-native';
+import { connect, Dispatch, MapStateToProps, MapDispatchToPropsFunction } from 'react-redux';
+import { View, Text, ScrollView, Dimensions, Animated } from 'react-native';
 
 import { EmployeesList } from './employees-list';
+import { OrganizationState } from '../reducers/organization/organization.reducer';
+import { Department } from '../reducers/organization/department.model';
+import { Employee } from '../reducers/organization/employee.model';
 import { AppState } from '../reducers/app.reducer';
+import { DepartmentsHScrollableList } from './departments/departments-horizontal-scrollable-list';
+import { DepartmentsTree } from './departments/departments-tree';
+import { DepartmentsTreeNode } from './departments/departments-tree-node';
+import { EmployeeMap, EmployeesStore } from '../reducers/organization/employees.reducer';
+import { EmployeeCardWithAvatar } from './employee-card-with-avatar';
+import { PeopleActions, updateDepartmentIdsTree } from '../reducers/people/people.action';
 
 interface PeopleCompanyProps {
     routeName: string;
+    headDepartment: Department;
+    departmentsTree: DepartmentsTree;
+    departmentIdsBranch?: string[];
 }
 
+interface PeopleCompanyDispatchProps {
+    updateDepartmentIdsTree: (index: number, departmentId: string) => void;
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<PeopleActions>) => ({
+    updateDepartmentIdsTree: (index: number, departmentId: string) => {
+        dispatch(updateDepartmentIdsTree(index, departmentId));
+    },
+});
+
 const mapStateToProps = (state: AppState): PeopleCompanyProps => ({
-    routeName: 'Company'
+    routeName: 'Company',
+    headDepartment: state.organization.departments.filter((department) => department.isHeadDepartment === true)[0],
+    departmentsTree: departmentsTreeFor(state.organization),
+    departmentIdsBranch: state.people.departmentIdsBranch
 });
 
-export class PeopleCompanyImpl extends React.Component<PeopleCompanyProps> {  
+function departmentsTreeFor(organization: OrganizationState) {
+    const topLevelDepartment: Department = organization.departments.filter((department) => department.isHeadDepartment === true)[0];
```

Comment:
it seems what it should be placed in the init state

Reply:
But... not init. Combine LoadEmployeeFinished and LoadDepartmentsFinished and emit your specific action, then handle it in your people reducer to make "init" tree